tests/Resources/assert.addontest.js

```diff
@@ -0,0 +1,1185 @@
+			});
+		});
+	});
+
+	describe('#fail()', () => {
+		it('is a function', () => {
+			assert.fail.should.be.a.Function;
+		});
+
+		it('throws with default error message with no arguments', () => {
+			// AssertionError [ERR_ASSERTION]: Failed
+			should.throws(function () {
+				assert.fail();
+			},
+			function (err) {
+				return err instanceof assert.AssertionError && err.message === 'Failed';
+			});
+		});
+
+		it('throws AssertionError with given message string', () => {
+			// AssertionError [ERR_ASSERTION]: boom
+			should.throws(function () {
+				assert.fail('boom');
+			},
+			function (err) {
+				return err instanceof assert.AssertionError && err.message === 'boom';
+			});
+		});
+
+		it('throws supplied Error', () => {
+			// TypeError: need array
+			should.throws(function () {
+				assert.fail(new TypeError('need array'));
+			},
+			function (err) {
+				return err instanceof TypeError && err.message === 'need array';
+			});
+		});
+	});
+
+	describe('#throws()', () => {
+		it('is a function', () => {
+			assert.throws.should.be.a.Function;
+		});
+
+		// TODO: Test for:
+		// When no `expected` argument is given, treat like we only care whether an Error is thrown (not validating it)
+
+		it('throws when function does not throw an Error', () => {
+			should.throws(function () {
+				assert.throws(
+					() => 1
+				);
+			},
+			function (error) {
+				return error instanceof assert.AssertionError && error.message === 'Missing expected exception.';
+			});
+		});
+
+		it('throws when first argument is not a function', () => {
+			should.throws(function () {
+				assert.throws(1, {});
+			},
+			function (error) {
+				return error instanceof TypeError && error.message === 'The "fn" argument must be of type Function. Received type number';
+			});
+		});
+
+		it('does not throw when validation function returns true', () => {
+			should.doesNotThrow(function () {
+				const err = new TypeError('Wrong value');
+				err.code = 404;
+				err.foo = 'bar';
+				err.info = {
+					nested: true,
+					baz: 'text'
+				};
+				err.reg = /abc/i;
+
+				assert.throws(
+					() => { throw err; },
+					function (error) {
+						return error === err;
+					}
+				);
+			});
+		});
+
+		it('does not throw when validation function is an arrow function and returns true', () => {
+			should.doesNotThrow(function () {
+				const err = new TypeError('Wrong value');
+				err.code = 404;
+				err.foo = 'bar';
+				err.info = {
+					nested: true,
+					baz: 'text'
+				};
+				err.reg = /abc/i;
+
+				assert.throws(
+					() => { throw err; },
+					error => error === err
+				);
+			});
+		});
+
+		it('throws when validation function returns false', () => {
+			const err = new TypeError('Wrong value');
+			err.code = 404;
+			err.foo = 'bar';
+			err.info = {
+				nested: true,
+				baz: 'text'
+			};
+			err.reg = /abc/i;
+			should.throws(function () {
+				assert.throws(
+					() => { throw err; },
+					function (error) {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/a...
  
  
  DangerID: danger-id-default;
  File: tests/Resources/assert.addontest.js;
  Line: 371;
-->

- :warning: tests/Resources/assert.addontest.js line 371 – 'error' is defined but never used. (no-unused-vars)